### ISSUE TITLE
[E047] Can't assign a value to unregistered extension attribute 'trf_data'. Did you forget to call the `set_extension` method?

### ISSUE THREAD

**Issue by Dan-Git-source** [explosion/spaCy#7356]
I'm trying to use Spacy's en_core_web_trf for sentence separation.
For some reason I'm getting "[E047] Can't assign a value to unregistered extension attribute 'trf_data'. Did you forget to call the `set_extension` method?" When I switched the model to en_core_web_lg it worked. Any idea how can I used the trf model?
Thanks

**Comment by svlandeg** [comment]
Hm, that's strange - we'd expect `spacy-transformers` to be setting this attribute correctly.

Can you share the code you executed when you ran into this, and the entire error stack trace? Can you also share the version of `spacy` and `spacy-transformers` that are installed in your environment?

**Comment by Dan-Git-source** [comment]
spacy==3.0.3, spacy-transformers==1.0.1, spacy-nightly==3.0.0rc5

error:
_RemoteTraceback                          Traceback (most recent call last)
_RemoteTraceback: 
"""
Traceback (most recent call last):
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/externals/loky/process_executor.py", line 431, in _process_worker
    r = call_item()
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/externals/loky/process_executor.py", line 285, in __call__
    return self.fn(*self.args, **self.kwargs)
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/_parallel_backends.py", line 595, in __call__
    return self.func(*args, **kwargs)
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/parallel.py", line 262, in __call__
    return [func(*args, **kwargs)
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/parallel.py", line 262, in <listcomp>
    return [func(*args, **kwargs)
  File "<ipython-input-7-298f9baa4640>", line 4, in extract_sentences_from_call
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/spacy/language.py", line 994, in __call__
    error_handler(name, proc, [doc], e)
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/spacy/util.py", line 1497, in raise_error
    raise e
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/spacy/language.py", line 989, in __call__
    doc = proc(doc, **component_cfg.get(name, {}))
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/spacy_transformers/pipeline_component.py", line 184, in __call__
    self.set_annotations([doc], outputs)
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/spacy_transformers/pipeline_component.py", line 238, in set_annotations
    doc._.trf_data = data
  File "/Users/daneran/opt/miniconda3/envs/test2/lib/python3.9/site-packages/spacy/tokens/underscore.py", line 60, in __setattr__
    raise AttributeError(Errors.E047.format(name=name))
AttributeError: [E047] Can't assign a value to unregistered extension attribute 'trf_data'. Did you forget to call the `set_extension` method?
"""

The above exception was the direct cause of the following exception:

AttributeError                            Traceback (most recent call last)
<ipython-input-7-298f9baa4640> in <module>
     98     K = batch_calls.shape[0]
     99 
--> 100     for res in Parallel(n_jobs=cpu_count(), prefer='processes'
    101                         )(delayed(extract_sentences_from_call)((i, batch_calls.iloc[i]))
    102                           for i in range(K)):

~/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/parallel.py in __call__(self, iterable)
   1052 
   1053             with self._backend.retrieval_context():
-> 1054                 self.retrieve()
   1055             # Make sure that we get a last message telling us we are done
   1056             elapsed_time = time.time() - self._start_time

~/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/parallel.py in retrieve(self)
    931             try:
    932                 if getattr(self._backend, 'supports_timeout', False):
--> 933                     self._output.extend(job.get(timeout=self.timeout))
    934                 else:
    935                     self._output.extend(job.get())

~/opt/miniconda3/envs/test2/lib/python3.9/site-packages/joblib/_parallel_backends.py in wrap_future_result(future, timeout)
    540         AsyncResults.get from multiprocessing."""
    541         try:
--> 542             return future.result(timeout=timeout)
    543         except CfTimeoutError as e:
    544             raise TimeoutError from e

~/opt/miniconda3/envs/test2/lib/python3.9/concurrent/futures/_base.py in result(self, timeout)
    438                 raise CancelledError()
    439             elif self._state == FINISHED:
--> 440                 return self.__get_result()
    441             else:
    442                 raise TimeoutError()

~/opt/miniconda3/envs/test2/lib/python3.9/concurrent/futures/_base.py in __get_result(self)
    387     def __get_result(self):
    388         if self._exception:
--> 389             raise self._exception
    390         else:
    391             return self._result

AttributeError: [E047] Can't assign a value to unregistered extension attribute 'trf_data'. Did you forget to call the `set_extension` method?

Can't share the code in it's entirty but here is the jist of it:
`orig_sentences=[sent.text.strip() for sent in nlp(document_body).sents]`
...
```
for res in Parallel(n_jobs=cpu_count(), prefer='processes'
                        )(delayed(extract_sentences_from_call)((i, batch_calls.iloc[i]))
                          for i in range(K)):
        i, sentences, locations, regions = res
```

**Comment by adrianeboyd** [comment]
After thinking about this some more, I think we should just modify `spacy-transformers` to set the extension on call+pipe rather than init. Users shouldn't have to manage this.

**Comment by adrianeboyd** [comment]
Should be fixed by https://github.com/explosion/spacy-transformers/pull/269

**Comment by adrianeboyd** [comment]
Fix now published in `spacy-transformers` v1.0.2
